barretenberg/cpp/src/barretenberg/stdlib_circuit_builders/ultra_circuit_builder.cpp

```diff
@@ -2863,32 +2863,31 @@ void UltraCircuitBuilder_<FF>::create_poseidon2_internal_gate(const poseidon2_in
     ++this->num_gates;
 }
 
-template <typename ExecutionTrace> uint256_t UltraCircuitBuilder_<ExecutionTrace>::hash_circuit()
+/**
```

Comment:
it will differ because the wire values will differ, right? maybe it would be useful to add a sentence here about why this function is useful

Reply:
The hash doesn't include any witness _values_, only the indices into the variables array. So its not that computing the circuit with different witness values will result in different hashes. Its just that in the dummy witness case we use `assert_equal` + `add_variable` to set the values of dummy witnesses, and those operations effect the content of the `real_variable_index` array. Eventually though the two result in an identical VK (i.e. an identical circuit). So this is still useful in all cases except the dummy witness case. (I'll update the comment).